aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -523,6 +523,19 @@ void mse_kernel(TensorIterator& iter) {
   });
 }
 
+bool equal_kernel(TensorIterator& iter) {
+  std::atomic<bool> equal{true};
```

Comment:
You are not reading equal inside of threads, no need to have it atomic.

Reply:
If you want to use equal as premature exit, you will need to use `loop` and `iter.for_each` instead of `cpu_kernel`. I recommend benchmarking to decide if it is needed.

Reply:
I did some benchmark, the result is at the top of the page. Maybe we should use premature exit, but I did not found some example codes like this.